src/net.cpp

```diff
@@ -2185,11 +2185,36 @@ void CConnman::WakeMessageHandler()
 
 void CConnman::ThreadDNSAddressSeed()
 {
+    constexpr int TARGET_OUTBOUND_CONNECTIONS = 2;
```

Comment:
Sorry for the late realization, but maybe this should be
```suggestion
    const int target_outbound_connections = std::clamp(args.GetIntArg("-maxconnections", DEFAULT_MAX_PEER_CONNECTIONS), 1, 2);
```
This would be done in order to minimize unnecessarily waiting for 30s if we don't allow more than 1 connection, but simultaneously wait for 2.

Make it at least 1 since we shouldn't get here if maxconnections is `<= 0` as that would turn off `-dnsseed` and `ThreadDNSAddressSeed()`.

Reply:
I kind of agree, even though that was already the case before this PR.

I'd like to hear your thoughts on this @mzumsande

Reply:
> I'd like to hear your thoughts on this @mzumsande

I think it doesn't really make sense to try and improve the user experience of `-maxconnection=1` - doing that is stupid and dangerous, it's basically begging to be eclipsed. If someone wants just one peer they should use `-connect` to at least pick one that they trust. It has been suggested before to disable a very low `-maxconnection` or at least print out a big warning.

Reply:
@mzumsande thanks for your informed opinion!

@sr-gi will try to go through the latest version in the next few days.